crypto/s2n_rsa.c

```diff
@@ -121,7 +121,7 @@ static int s2n_rsa_encrypt(const struct s2n_pkey *pub, struct s2n_blob *in, stru
     /* Safety: RSA_public_encrypt does not mutate the key */
     int r = RSA_public_encrypt(in->size, (unsigned char *) in->data, (unsigned char *) out->data,
             s2n_unsafe_rsa_get_non_const(pub_key), RSA_PKCS1_PADDING);
-    S2N_ERROR_IF(r != out->size, S2N_ERR_SIZE_MISMATCH);
+    POSIX_ENSURE((int64_t) r == out->size, S2N_ERR_SIZE_MISMATCH);
```

Comment:
Sorry if I'm missing something here, but if `r` is cast to `int64` but `out->size` is unsigned, won't `r` still be cast to unsigned when comparing? And shouldn't the compiler still complain about this with the sign-compare check?

Reply:
ah right; yeah they both need to be cast to `int64_t`, otherwise the check will complain.